gamemodes/terrortown/entities/weapons/weapon_tttbase.lua

```diff
@@ -1045,12 +1050,62 @@ function SWEP:Initialize()
 	end
 end
 
+local idle_activities = {
+	[ACT_VM_IDLE] = true,
+	[ACT_VM_IDLE_TO_LOWERED] = true,
+	[ACT_VM_IDLE_LOWERED] = true,
+	[ACT_VM_IDLE_SILENCED] = true,
+	[ACT_VM_IDLE_EMPTY_LEFT] = true,
+	[ACT_VM_IDLE_EMPTY] = true,
+	[ACT_VM_IDLE_DEPLOYED_EMPTY] = true,
+	[ACT_VM_IDLE_8] = true,
+	[ACT_VM_IDLE_7] = true,
+	[ACT_VM_IDLE_6] = true,
+	[ACT_VM_IDLE_5] = true,
+	[ACT_VM_IDLE_4] = true,
+	[ACT_VM_IDLE_3] = true,
+	[ACT_VM_IDLE_2] = true,
+	[ACT_VM_IDLE_1] = true,
+	[ACT_VM_IDLE_DEPLOYED] = true,
+	[ACT_VM_IDLE_DEPLOYED_8] = true,
+	[ACT_VM_IDLE_DEPLOYED_7] = true,
+	[ACT_VM_IDLE_DEPLOYED_6] = true,
+	[ACT_VM_IDLE_DEPLOYED_5] = true,
+	[ACT_VM_IDLE_DEPLOYED_4] = true,
+	[ACT_VM_IDLE_DEPLOYED_3] = true,
+	[ACT_VM_IDLE_DEPLOYED_2] = true,
+	[ACT_VM_IDLE_DEPLOYED_1] = true,
+	[ACT_VM_IDLE_M203] = true,
+}
+local function IsIdleActivity(vm)
+	return idle_activities[ vm:GetSequenceActivity( vm:GetSequence() ) ] or false
+end
+local function IsCounterStrikeWeapon(str)
+	return string.StartsWith(string.lower(str):gsub("\\","/"), "models/weapons/cstrike/")
+end
+
+local silenced_weapons = {
+	weapon_ttt_sipistol = true,
+	weapon_ttt_tmp_s = true,
+	weapon_ttt_m4a1_s = true,
+}
```

Comment:
I'm not sure if I like that in the weapon base. Shouldn't that be part of the weapon itself? As `SWEP.silenced = true` or something similar

Reply:
this is temporary to check that I have it solved; the correct way to handle this would be to use `SWEP.IdleAnim = ACT_VM_IDLE_SILENCED` since that's what the silenced variants of the base weapon animations use

Reply:
ah, so that is why it is a draft then, gotcha. Thanks for working on this!